.NET/Microsoft.Recognizers.Text.DateTime/Extractors/BaseTimeZoneExtractor.cs

```diff
@@ -51,7 +51,35 @@ private IEnumerable<Token> MatchLocationTimes(string text)
 
             var timeMatch = config.LocationTimeSuffixRegex.Matches(text);
```

Comment:
This cost is being paid even if other entities ending in "time" were already extracted. Can't this be improved?